vm/rust/src/lib.rs

```diff
@@ -220,8 +241,22 @@ pub extern "C" fn cairoVMCall(
                     JunoAppendResponse(reader_handle, felt_to_byte_array(&data).as_ptr());
                 }
             }
+            // We only need to return the state_diff when creating the genesis state_diff.
+            // Calling this for all other RPC requests is a waste of resources.
+            if return_state_diff == 1 {
+                match state.to_state_diff() {
+                    Ok(state_diff) => {
+                        let json_state_diff =
+                            jsonrpc::StateDiff::from_state_maps(&state_diff.state_maps);
+                        append_state_diff(reader_handle, &json_state_diff, &mut writer_buffer);
+                    }
```

Comment:
I don't like the idea of having flags to signal stuff that should happen or not. I believe this functions are due for refactoring but we will address that later on. Just signaling here